cca.go

```diff
@@ -0,0 +1,118 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package stat
+
+import (
+	"errors"
+	"math"
+
+	"github.com/gonum/floats"
+	"github.com/gonum/matrix"
+	"github.com/gonum/matrix/mat64"
+)
+
+// CanonicalCorrelations returns the canonical correlations, corr,
+// between the corresponding pairs of canonical variables. Canonical
+// variables can be obtained by projecting the sphered data into the
+// left and right eigenvectors (pvecs and qvecs) of the canonical
+// correlation matrix. The canonical correlation matrix is calculated
+// using the singular value decompositions of the centered input.
+// These singular value decompositions, xsvd and ysvd, are also returned.
+//
+// The slice weights is used to weight the observations. If weights is nil,
+// each weight is considered to have a value of one, otherwise the length of
+// weights must match the number of observations or CanonicalCorrelations will
+// panic.
+//
+// CanonicalCorrelations will panic if the inputs x and y do not have the same
+// number of rows, and their columns should be interepretable as two different
+// sets of measurements on the same observations.
+//
+// The sphered data are Xc * Sx^(-1/2) and Yc * Sy^(-1/2) respectively, and the
+// correlation matrix between the sphered data is the canonical correlation matrix,
+// Sx^(-1/2) * Sxy * Sy^(-1/2).
+// 	In cases where S^(-1/2) is ambiguous, S^(-1/2) is taken to be E * D^(-1/2) * E'
```

Comment:
Honestly, me too. Maybe I'll just re-arrange it to have the notation first and equations second.